src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -88,19 +71,13 @@ p5.RendererGL.prototype.vertex = function(x, y) {
     u = arguments[3];
     v = arguments[4];
   }
-  var vert = new p5.Vector(x, y, z);
-  this.immediateMode.vertices.push(vert);
-  var vertexColor = this.curFillColor || [0.5, 0.5, 0.5, 1.0];
-  this.immediateMode.vertexColors.push(
-    vertexColor[0],
-    vertexColor[1],
-    vertexColor[2],
-    vertexColor[3]
-  );
-
-  this.immediateMode.uvCoords.push(u, v);
 
-  return this;
+  var im = this.immediateMode;
+  var push = Array.prototype.push;
+  im.vertices.push(new p5.Vector(x, y, z));
+  push.apply(im.vertexColors, this._diffuseColor);
+  push.apply(im.vertexAmbients, this._ambientColor);
```

Comment:
this is purely a performance optimization that i made after profiling this code. the 2nd & 3rd calls are equivalient to:
````
Array.prototype.push(im.vertexColors, this._diffuseColor);
Array.prototype.push(im.vertexAmbients, this._ambientColor);
````
which appends the elements of the color arrays onto the vertex buffer arrays.

the `push` temporary is just there to save accessing the push method repeatedly. this was more of an issue originally when this code handled specularity, normals, ambients, etc...

Reply:
I understand what the code does, but I don't understand why the method isn't invoked the same way for all 4 cases. It's a strange thing to encounter side-by-side; a note about the optimization and why it's only needed in half of the cases would be helpful.

Reply:
The four cases are not the same, `push.apply` concatenates all the elements of an array onto the first argument whereas `push` just adds one element to an array:
```javascript
var a = [];
Array.prototype.push.apply(a, [1,2,3]);
console.log(a); // [1, 2, 3]
var b = [];
b.push([1, 2, 3]);
console.log(b); // [[1, 2, 3]]
```

It's equivalent to the spread operator here:
```javascript
var c = [];
c.push(...[1, 2, 3]);
console.log(c);
```
But we aren't using newer language features, so we have to use `push.apply`

Reply:
it's only invoked like that in those two cases because those are the only two cases where the input elements are in an array.

Reply:
spread also allocates a new array. which would be very bad here.

Reply:
got it, thank you.

Reply:
there still doesn't seem to be any reason that this can't be:

```
push.apply(im.vertices, new p5.Vector(x, y, z));
push.apply(im.vertexColors, this._diffuseColor);
push.apply(im.vertexAmbients, this._ambientColor);
push.apply(im.uvs, [u, v]);
```
unless the creation of the array on the last line is a performance hit? but could that just be noted with a comment? otherwise, interleaving the style choices still looks strange to me.

Reply:
1) `push.apply(im.vertices, new p5.Vector(x, y, z))` is an invalid invocation of the `apply` method. the second argument needs to be an array
2) this is correct, but requires creating a new array - exactly what i'm trying to avoid here.